iceoryx_utils/include/iceoryx_utils/internal/cxx/unique_ptr.inl

```diff
@@ -0,0 +1,160 @@
+
+#ifndef IOX_UTILS_CXX_UNIQUE_PTR_INL
+#define IOX_UTILS_CXX_UNIQUE_PTR_INL
+
+#include "iceoryx_utils/cxx/unique_ptr.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+template <typename T>
+unique_ptr<T>::unique_ptr(std::function<void(T*)>&& deleter) noexcept
+    : m_deleter(deleter)
+{
+}
+
+template <typename T>
+unique_ptr<T>::unique_ptr(ptr_t ptr, std::function<void(T*)>&& deleter) noexcept
+    : m_ptr(ptr)
+    , m_deleter(std::move(deleter))
+{
+}
+
+template <typename T>
+unique_ptr<T>::unique_ptr(std::nullptr_t) noexcept
+{
+    reset();
```

Comment:
Why call reset here? The members have been already initialized to an reset state in the header.

Reply:
This constructor is used when assigning a unique_ptr instance to nullptr, therefore the previous instance must be reset (in case its contents have not already been moved / deleted).
I think you saw this later as you mention it below.